### PR TITLE
Auth

### LINKED ISSUE #14: Mise en place de la page d'authentification

### PR DESCRIPTION
**Pull request by yannmtrn**
Adding signin page, signup page and traitment of these

- Auth.php renamed from Register.php
- Adding username, firstname and lastname on pandanim.sql and dataDictionnary.md
- Adding init session
- Other small fixed

### CODE REVIEW

File: database/pandanim.sql [hunk 1]
```diff
@@ -102,6 +102,9 @@ CREATE TABLE IF NOT EXISTS `anime_studio` (
 -- DROP TABLE IF EXISTS `user`;
 CREATE TABLE IF NOT EXISTS `user` (
   `id` smallint(5) UNSIGNED NOT NULL AUTO_INCREMENT,
+  'username' varchar(30) NOT NULL,
+  'firstname' varchar(40) NOT NULL,
+  'lastname' varchar(40) NOT NULL,
```
Comment (lbanitz-net) [inline database/pandanim.sql]: wrong caracter : replaced ' with `

File: database/pandanim.sql [hunk 2]
```diff
@@ -102,6 +102,9 @@ CREATE TABLE IF NOT EXISTS `anime_studio` (
...
   `email` varchar(255) NOT NULL,
   `password` varchar(128) NOT NULL COMMENT 'hashed (SHA-512)',
   `roles` json NOT NULL,
```
Comment (lbanitz-net) [inline database/pandanim.sql]: added a default value

File: app/Config/App.php [hunk 3]
```diff
@@ -24,7 +24,7 @@ class App extends BaseConfig
      *
      * @var string
      */
-    public $baseURL = 'http://localhost:8080/';
+    public $baseURL = 'http://localhost:8888/';
```
Comment (lbanitz-net) [inline app/Config/App.php]: removed default base url -> defined in .env

File: app/Controllers/Auth.php [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+        // On donne des règles de validation
+        $validation->setRules([
+            'email' => [
+                'rules'  => 'required|valid_email',
+                'errors' => [
+                    'required' => 'Le {field} est obligatoire',
+                ],
+            ],
+            'password' => [
+                'rules'  => 'required|min_length[8]',
```
Comment (lbanitz-net) [inline app/Controllers/Auth.php:42]: attention, la vérificaton de sécurité de mot de passe me semble un peu light

Comment (lbanitz-net) [reply app/Controllers/Auth.php:42]: Nouvelle issue ajoutée : https://github.com/CESI-CDA-2022-23-Groupe-4/pandanim/issues/27

**Review by FBanitz — APPROVED** [review]

**Pull request merged**